### PR TITLE
Updates to acceptance tests

### PR DESCRIPTION
**Pull request by dbelcham**
Related to #4457 

Renamed a bunch of tests for clarity, cleaned up a bunch of old comments, updated functionality of one test

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj [hunk 1]
```diff
@@ -40,6 +40,11 @@
     <_PackageFiles Remove="**\obj\**\*.cs" />
     <_PackageFiles Remove="Core\**\*.cs" />
     <_PackageFiles Remove="AssemblyInfo.cs" />
+    <_PackageFiles Remove="Sagas\When_correlated_property_value_is_changed.cs" />
```
Comment (bording) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: Why have these entries been added?

File: src/NServiceBus.AcceptanceTests/Sagas/When_correlated_property_value_is_changed.cs [hunk 2]
```diff
@@ -21,10 +21,9 @@ await Scenario.Define<Context>()
                         {
                             DataId = Guid.NewGuid()
                         })))
-                    .Done(c => c.FailedMessages.Any())
+                    .Done(c => c.ModifiedCorrelationProperty)
```
Comment (bording) [inline src/NServiceBus.AcceptanceTests/Sagas/When_correlated_property_value_is_changed.cs]: Does this change actually have the desired outcome? Are we now racing to shutdown before actually finishing processing the saga and getting the exception thrown?

File: src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj (hunk 1 above)
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/NServiceBus.AcceptanceTests.csproj]: I think this is done automatically by VS for some reason. should be removed though.

File: src/NServiceBus.AcceptanceTests/Sagas/When_correlated_property_value_is_changed.cs (hunk 2 above)
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Sagas/When_correlated_property_value_is_changed.cs]: I think it should be no issue, as the att framework should wait for "incomplete" message to be processed (to either succeed or fail).

**Pull request merged**